src/parser/common.rs

```diff
@@ -201,3 +210,34 @@ pub fn skip_newlines_and_comments<'a>(input: ParserInput<'a>) -> ParserResult<'a
     )))(input)?;
     Ok((input, ()))
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::{expression::Expression, instruction::MemoryReference, parser::lex, real};
+
+    use super::parse_waveform_invocation;
+
+    #[test]
+    fn waveform_invocation() {
+        let input = "wf(a: 1.0, b: %var, c: ro[0])";
+        let lexed = lex(input);
+        let (remainder, waveform) = parse_waveform_invocation(&lexed).unwrap();
+        assert_eq!(remainder, &[]);
+        assert_eq!(
+            waveform.parameters,
+            vec![
+                ("a".to_owned(), Expression::Number(real!(1f64))),
+                ("b".to_owned(), Expression::Variable("var".to_owned())),
+                (
+                    "c".to_owned(),
+                    Expression::Address(MemoryReference {
+                        name: "ro".to_owned(),
+                        index: 0
+                    })
+                )
+            ]
+            .into_iter()
+            .collect()
+        )
+    }
+}
```

Comment:
Nicely tested, thanks!

I was contemplating trying to add some property-based testing library like [`proptest`](https://github.com/AltSysrq/proptest) to generate round-trip tests like `assert_eq!(program, parse(to_string(program))`. An idea for a different PR.

Reply:
Sure, I'd be interested in seeing what that'd look like here